src/test/kotlin/uk/gov/justice/digital/hmpps/breachnoticeapi/integration/DomainEventTest.kt

```diff
@@ -0,0 +1,204 @@
+        .bodyValue(BreachNotice(crn = "X000111"))
+        .exchange()
+        .expectStatus()
+        .isCreated
+
+      val breachNotice = breachNoticeRepository.findByCrn("X000111").single()
+      assertThat(breachNotice.crn).isEqualTo("X000111")
+      assertThat(breachNotice.id).isNotNull()
+
+      breachNotice.completedDate = LocalDateTime.now()
+      breachNoticeRepository.save(breachNotice)
+
+      val message: String = "{\"eventType\":\"probation-case.merge.completed\",\"version\":1,\"occurredAt\":\"2025-03-04T10:30:07.329287Z\",\"description\":\"A merge has been completed on the probation case\",\"additionalInformation\":{\"sourceCRN\":\"X000111\",\"targetCRN\":\"X000102\"},\"personReference\":{\"identifiers\":[{\"type\":\"CRN\",\"value\":\"X000102\"}]}}\n"
+
+      val responseFuture = inboundSnsClient.publish(
+        PublishRequest.builder().topicArn("arn:aws:sns:eu-west-2:000000000000:hmppsbreachnoticetopic").message(message).messageAttributes(
+          mapOf("eventType" to MessageAttributeValue.builder().dataType("String").stringValue("probation-case.merge.completed").build()),
+        ).build(),
+      )
+      val response = responseFuture.get(10, TimeUnit.SECONDS)
+
+      assertThat(response.messageId()).isNotNull()
+
+      Awaitility.await().atMost(10, TimeUnit.SECONDS).untilAsserted {
+        val breachNoticeUpdated: BreachNoticeEntity = breachNoticeRepository.findById(breachNotice.id).orElse(null)
+        assertThat(breachNoticeUpdated).isNotNull
+        assertThat(breachNoticeUpdated.crn).isEqualTo("X000111")
+        assertThat(breachNoticeUpdated.id).isNotNull()
+        assertThat(breachNoticeUpdated.reviewRequiredDate).isNull()
+        assertThat(breachNoticeUpdated.reviewEvent).isNull()
+      }
+    }
+
+    @Test
+    fun `unmerge event should update CRN for active breach notice`() {
+      webTestClient.post()
+        .uri("/breach-notice")
+        .headers(setAuthorisation(roles = listOf("ROLE_BREACH_NOTICE")))
+        .bodyValue(BreachNotice(crn = "X000121"))
+        .exchange()
+        .expectStatus()
+        .isCreated
+
+      val breachNotice = breachNoticeRepository.findByCrn("X000121").single()
+      assertThat(breachNotice.crn).isEqualTo("X000121")
+      assertThat(breachNotice.id).isNotNull()
+
+      val message: String = "{\n" +
+        "  \"eventType\":\"probation-case.unmerge.completed\",\n" +
+        "  \"version\":1,\n" +
+        "  \"occurredAt\":\"2025-03-03T12:20:13.6147Z\",\n" +
+        "  \"description\":\"An unmerge has been completed on the probation case\",\n" +
+        "  \"additionalInformation\":{\n" +
+        "    \"reactivatedCRN\":\"X000103\",\n" +
+        "    \"unmergedCRN\":\"X000121\"},\n" +
+        "  \"personReference\":{\n" +
+        "    \"identifiers\":[\n" +
+        "      {\n" +
+        "        \"type\":\"CRN\",\n" +
+        "        \"value\":\"X000121\"\n" +
+        "      }\n" +
+        "    ]\n" +
+        "  }\n" +
+        "}"
```

Comment:
Minor thing - but you could use a multi-line string for these which would mean you don't need to escape everything.  Also you can add a language comment so IntelliJ highlights and formats it as JSON.
```suggestion
      // language=json
      val message = """{
        "eventType":"probation-case.unmerge.completed",
        "version":1,
        "occurredAt":"2025-03-03T12:20:13.6147Z",
        "description":"An unmerge has been completed on the probation case",
        "additionalInformation":{
          "reactivatedCRN":"X000103",
          "unmergedCRN":"X000121"
        },
        "personReference":{
          "identifiers":[
            {
              "type":"CRN",
              "value":"X000121"
            }
          ]
        }
      }""".trimIndent()
```